confluence_utils/cli.py

```diff
@@ -1,39 +1,188 @@
-    envvar=["CONFLUENCE_SPACE"],
-)
-@click.option(
-    "--token",
-    required=True,
-    help="Confluence API Token. Optionally set with CONFLUENCE_TOKEN.",
-    envvar=["CONFLUENCE_TOKEN"],
-)
+@client_options
 @click.argument("path", type=click.Path(exists=True, resolve_path=True))
 def publish(path: str, url: str, space: str, token: str) -> None:
-    if os.path.isfile(path):
-        click.echo(f"publishing file: {path}")
+    confluence = Confluence(
+        url=url,
+        token=token,
+    )
+
+    if os.path.isfile(path) and path.endswith(".md"):
+        publish_file(path, space, confluence)
+    elif os.path.isdir(path):
+        for file in get_files(path, ".md"):
+            publish_file(file, space, confluence)
+
+
+def publish_file(path: str, space: str, confluence: Confluence) -> None:
+    markdown_file = MarkdownFile.from_path(path, space)
+
+    if markdown_file.parent_file_path is not None:
+        parent_path = os.path.join(
+            os.path.dirname(path),
+            markdown_file.parent_file_path,
+        )
+        if not os.path.isfile(parent_path):
+            click.echo(f"parent page does not exist skipping: {path}")
+            return
+
+        parent_file = MarkdownFile.from_path(parent_path, space)
+        # maybe add space validation here
+        if parent_file.page_id is not None:
+            markdown_file.parent_id = parent_file.page_id
+            markdown_file.update_front_matter(space)
+        elif parent_file.page_id is None:
+            # check if the page exist but the page id is not set
+            if confluence.page_exists(space=space, title=parent_file.title):
+                parent_id = confluence.get_page_id(space, parent_file.title)
+                parent_file.page_id = parent_id
+                markdown_file.parent_id = parent_id
+                markdown_file.update_front_matter(space)
+                parent_file.update_front_matter(space)
+            else:
+                click.echo(f"parent page does not exist skipping: {path}")
+                return
+
+    body, attachments = markdown_file.render_confluence_content()
+    page_with_title_exist = confluence.page_exists(
+        space=space, title=markdown_file.title
+    )
+
+    if not markdown_file.page_id and page_with_title_exist:
+        page_id = confluence.get_page_id(space, markdown_file.title)
+        markdown_file.page_id = page_id
+        markdown_file.update_front_matter(space)
+        update_page_response = confluence.update_page(
+            page_id=markdown_file.page_id,
+            title=markdown_file.title,
+            body=body,
+            parent_id=markdown_file.parent_id,
+        )
+        click.echo(
+            f"Updated page with page_id {update_page_response.get('id')}"
+        )
+    elif markdown_file.page_id and page_with_title_exist:
+        # look up page id to make sure confluence is aware of it
+        checkedSpace = None
```

Comment:
@ttrout55 my IDE is telling me `Variable in function should be lowercase` here. I guess it makes sense - we're using [snake_case](https://en.wikipedia.org/wiki/Snake_case) everywhere else in the project.

Reply:
yea my bad brain went java/go